### DISCUSSION TITLE
How to start a transaction that will use both welds & sqlx queries?

### DISCUSSION THREAD

**Discussion by NexRX** [weldsorm/welds#18] [Q&A] [answered]
So I'm trying to mix both welds and sqlx but I'm struggling because 
a) I cant use the welds tranactions for sqlx
b) I cant map a sqlx transaction into a welds one

Is there a way to do this and im just dumb as door nails? :D

**Comment by NexRX** [comment]
@lex148 Sorry to ping you all, but could you help me with this? :)

**Comment by lex148** [comment, marked as answer]
Hey @NexRX 

I am so sorry I didn't see your comment here until now. :( 

unfortunately they are not interchangeable. This is because they also need to work with `Tiberius` . I know that is probably not what you wanted to hear. 

The best workaround I can think of is make a welds connection from an existing sqlx connection. The sqlx connection is already wrapped in an Arc so no need to worry about copying/cloning it. you can just throw the welds connection way when you finish.

If it helps here is an example of using transactions in welds.
https://github.com/weldsorm/welds/blob/main/welds/examples/transactions.rs
